File: leetcode3151/solution.go
/*
Solutioni:

Time: O(n) | Space: O(1)

Runtime: 0 ms | 100.00%
Memory: 4.66 MB | 82.93%

- n: length of nums
*/

func isArraySpecial(nums []int) bool {
	for i := 1; i < len(nums); i++ {
		if (nums[i-1]&1)^(nums[i]&1) == 0 {
			return false
		}
	}

	return true
}

File: leetcode3151/solution.rs
/*
Solutioni:

Time: O(n) | Space: O(1)

Runtime: 0 ms | 100.00%
Memory: 2.27 MB | 85.71%

- n: length of nums
*/

impl Solution {
    pub fn is_array_special(nums: Vec<i32>) -> bool {
        for i in 1..nums.len() {
            if (nums[i - 1] & 1) ^ (nums[i] & 1) == 0 {
                return false;
            }
        }

        true
    }
}